packaging/os/redhat_repositories.py

```diff
@@ -0,0 +1,269 @@
+                    # Remember the name for later processing
+                    repo=RhsmRepository(self.module, _name=value)
+                    repos.append(repo)
+                    repos[-1].__setattr__(key, value)
+                elif repos:
+                    # Associate value with most recently recorded repo
+                    repos[-1].__setattr__(key, value)
+             
+        return repos
+
+    def filter(self, state=None):
+        ''' 
+            Return a list of Repositories where state matches (null: all, enabled, disabled)
+        '''
+
+        if self.repos is None:
+            self.repos = self._load_repo_list()
+
+            if state == 'all' or state is None:
+                for repo in self.repos:
+                    yield repo 
+
+            elif state == 'enabled':
+                for repo in self.repos:
+                    if repo.is_enabled():
+                        yield repo
+
+            elif state == 'disabled':
+                for repo in self.repos:
+                    if repo.is_disabled():
+                        yield repo
+
+    def enablerepo(self, name):
+        if len(name) > 0:
+            args="subscription-manager repos "
+            for n in name:
+                args+=" --enable="+n
+            rc, stdout, stderr = self.module.run_command(args, check_rc=True)
+
+    def disablerepo(self, name):
+        if len(name) > 0:
+            args="subscription-manager repos "
+            for n in name:
+                args += " --disable=" + n 
+            rc, stdout, stderr = self.module.run_command(args, check_rc=True)
+
+    def disable_all(self):
+        args = "subscription-manager repos --disable=*"
+        rc, stdout, stderr = self.module.run_command(args, check_rc=True)
+
+def list_stuff(rhsmrepos, stuff):
+    l=[]
+    for repo in rhsmrepos.filter(stuff):
+        l.append(dict(name=repo._name, enabled=repo.is_enabled))
+    return l
+
+def main():
+    #
+    # Initialise Ansible module
+    #
+    module = AnsibleModule(
+                argument_spec = dict(
+                    name    = dict(default=None, required=False, type="list"),
+                    state   = dict(default=None, choices=['enabled','disabled']),
+                    list    = dict(default=None, required=False),
+                ),
+                required_one_of = [['name', 'list']],
+                mutually_exclusive = [['name', 'list']],
+                supports_check_mode = False,
+            )
+
+    #
+    # Initialize RhsmRepositories
+    #
+    rhsmrepos = RhsmRepositories(module)
+
+    list = module.params['list']
+    name = module.params['name']
+    state = module.params['state']
+    #
+
+    if list:
+        result=list_stuff(rhsmrepos, state)         # FIXME : maybe review state usage here
+        module.exit_json(changed=False,repos=result)
+
+    elif name is not None:
+        if name == '*':
+            rhsmrepos.disable_all()
```

Comment:
@jfenal when i tried with the latest copy from your repo (https://raw.githubusercontent.com/jfenal/ansible-modules-jfenal/master/packaging/os/redhat_repositories.py) I got this: 

```
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):                                                                                        
  File "/usr/lib/python2.7/site-packages/ansible/executor/task_executor.py", line 119, in run                             
    res = self._execute()                                                                                                 
  File "/usr/lib/python2.7/site-packages/ansible/executor/task_executor.py", line 490, in _execute                        
    result = self._handler.run(task_vars=variables)                                                                       
  File "/usr/lib/python2.7/site-packages/ansible/plugins/action/normal.py", line 33, in run                               
    results = merge_hash(results, self._execute_module(tmp=tmp, task_vars=task_vars))                                     
  File "/usr/lib/python2.7/site-packages/ansible/plugins/action/__init__.py", line 565, in _execute_module                
    (module_style, shebang, module_data, module_path) = self._configure_module(module_name=module_name, module_args=module_args, task_vars=task_vars)                                                                                               
  File "/usr/lib/python2.7/site-packages/ansible/plugins/action/__init__.py", line 153, in _configure_module              
    task_vars=task_vars, module_compression=self._play_context.module_compression)                                        
  File "/usr/lib/python2.7/site-packages/ansible/executor/module_common.py", line 772, in modify_module                   
    (module_data, module_style, shebang) = _find_snippet_imports(module_name, module_data, module_path, module_args, task_vars, module_compression)                                                                                                 
  File "/usr/lib/python2.7/site-packages/ansible/executor/module_common.py", line 640, in _find_snippet_imports           
    recursive_finder(module_name, module_data, py_module_names, py_module_cache, zf)                                      
  File "/usr/lib/python2.7/site-packages/ansible/executor/module_common.py", line 441, in recursive_finder                
    tree = ast.parse(data)                                                                                                
  File "/usr/lib64/python2.7/ast.py", line 37, in parse                                                                   
    return compile(source, filename, mode, PyCF_ONLY_AST)                                                                 
  File "<unknown>", line 362                                                                                              
    notice( "redhat_repository: in main | list=" + str(p_list) + " | id=" + str(p_id) + " | state=" + str(p_state) + " |  
                                                                                                                       ^  
SyntaxError: EOL while scanning string literal                                                                            

fatal: [sat-cap-config-1.rhc-lab.iad.redhat.com]: FAILED! => {
    "failed": true,                                                                                                       
    "msg": "Unexpected failure during module execution.",                                                                 
    "stdout": ""                                                                                                          
}                                                                                                                         
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/ansible/executor/task_executor.py", line 119, in run
    res = self._execute()
  File "/usr/lib/python2.7/site-packages/ansible/executor/task_executor.py", line 490, in _execute
    result = self._handler.run(task_vars=variables)
  File "/usr/lib/python2.7/site-packages/ansible/plugins/action/normal.py", line 33, in run
    results = merge_hash(results, self._execute_module(tmp=tmp, task_vars=task_vars))
  File "/usr/lib/python2.7/site-packages/ansible/plugins/action/__init__.py", line 565, in _execute_module
    (module_style, shebang, module_data, module_path) = self._configure_module(module_name=module_name, module_args=module_args, task_vars=task_vars)
  File "/usr/lib/python2.7/site-packages/ansible/plugins/action/__init__.py", line 153, in _configure_module
    task_vars=task_vars, module_compression=self._play_context.module_compression)
  File "/usr/lib/python2.7/site-packages/ansible/executor/module_common.py", line 772, in modify_module
    (module_data, module_style, shebang) = _find_snippet_imports(module_name, module_data, module_path, module_args, task_vars, module_compression)
  File "/usr/lib/python2.7/site-packages/ansible/executor/module_common.py", line 640, in _find_snippet_imports
    recursive_finder(module_name, module_data, py_module_names, py_module_cache, zf)
  File "/usr/lib/python2.7/site-packages/ansible/executor/module_common.py", line 441, in recursive_finder
    tree = ast.parse(data)
  File "/usr/lib64/python2.7/ast.py", line 37, in parse
    return compile(source, filename, mode, PyCF_ONLY_AST)
  File "<unknown>", line 362
    notice( "redhat_repository: in main | list=" + str(p_list) + " | id=" + str(p_id) + " | state=" + str(p_state) + " |
                                                                                                                       ^
SyntaxError: EOL while scanning string literal

fatal: [sat-cap-content-1.rhc-lab.iad.redhat.com]: FAILED! => {
    "failed": true, 
    "msg": "Unexpected failure during module execution.", 
    "stdout": ""
}
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/ansible/executor/task_executor.py", line 119, in run
    res = self._execute()
  File "/usr/lib/python2.7/site-packages/ansible/executor/task_executor.py", line 490, in _execute
    result = self._handler.run(task_vars=variables)
  File "/usr/lib/python2.7/site-packages/ansible/plugins/action/normal.py", line 33, in run
    results = merge_hash(results, self._execute_module(tmp=tmp, task_vars=task_vars))
  File "/usr/lib/python2.7/site-packages/ansible/plugins/action/__init__.py", line 565, in _execute_module
    (module_style, shebang, module_data, module_path) = self._configure_module(module_name=module_name, module_args=module_args, task_vars=task_vars)
  File "/usr/lib/python2.7/site-packages/ansible/plugins/action/__init__.py", line 153, in _configure_module
    task_vars=task_vars, module_compression=self._play_context.module_compression)
  File "/usr/lib/python2.7/site-packages/ansible/executor/module_common.py", line 772, in modify_module
    (module_data, module_style, shebang) = _find_snippet_imports(module_name, module_data, module_path, module_args, task_vars, module_compression)
  File "/usr/lib/python2.7/site-packages/ansible/executor/module_common.py", line 640, in _find_snippet_imports
    recursive_finder(module_name, module_data, py_module_names, py_module_cache, zf)
  File "/usr/lib/python2.7/site-packages/ansible/executor/module_common.py", line 441, in recursive_finder
    tree = ast.parse(data)
  File "/usr/lib64/python2.7/ast.py", line 37, in parse
    return compile(source, filename, mode, PyCF_ONLY_AST)
  File "<unknown>", line 362
    notice( "redhat_repository: in main | list=" + str(p_list) + " | id=" + str(p_id) + " | state=" + str(p_state) + " |
                                                                                                                       ^
SyntaxError: EOL while scanning string literal

fatal: [sat-cap-infra-1.rhc-lab.iad.redhat.com]: FAILED! => {
    "failed": true, 
    "msg": "Unexpected failure during module execution.", 
    "stdout": ""
}
```